### PR TITLE
test(web): add unit tests to components [VIZ-1465]

### PR DESCRIPTION
**Pull request by mulengawilfred**
# Overview
Added unit tests to the following components under web/src/beta/lib/reearth-ui/components:
- Panel
- Popup
- Popup menu
- Popup panel
- Radio
- Radio group
- Range slider
- Range selector
- Slide
- Slider
- Switcher
- Tabs
- Text area
- Text input
- Time picker
- Tooltip
- Typography
## What I've done
Test coverage is currently 43.43% from 41.56% on main branch.
## Summary by CodeRabbit

- **Tests**
  - Added comprehensive test suites for various UI components, including Panel, Popup, PopupMenu, PopupPanel, Radio, RadioGroup, RangeSlider, Selector, Slide, Slider, Switcher, Tabs, TextArea, TextInput, TimePicker, Tooltip, and Typography.
  - Tests cover rendering, user interactions, callback handling, accessibility attributes, visual states, and edge cases for each component.
  - Minor updates to components to support testing (e.g., adding test IDs and standardizing test identifiers).

### CODE REVIEW

File: web/src/beta/lib/reearth-ui/components/Slider/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    expect(sliderElement).toHaveAttribute("aria-valuemin", "0");
+    expect(sliderElement).toHaveAttribute("aria-valuemax", "10");
+  });
+
+  test("renders in disabled state", () => {
+    render(<Slider value={50} disabled />);
+
+    const sliderElement = screen.getByRole("slider");
+    expect(sliderElement).toHaveAttribute("aria-disabled", "false");
+  });
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Slider/index.test.tsx]: Thanks for spotting this. After investigation, looks like even after passing the `data-testid` to the component, the `disabled` prop is used to conditionally render styles and not applied to `aria-disabled` prop. Will skip this one for now.

File: web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  });
+
+  test("updates when value prop changes", () => {
+    const { rerender } = render(<RangeSlider value={[20, 80]} />);
+
+    rerender(<RangeSlider value={[30, 70]} />);
+
+    const sliderElement = document.querySelector(".rc-slider");
+    expect(sliderElement).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx]: Thanks. This has been added.

File: web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    const disabledSlider = document.querySelector(".rc-slider-disabled");
+    expect(disabledSlider).toBeInTheDocument();
+  });
+
+  test("uses calculated step when not explicitly provided", () => {
+    render(<RangeSlider value={[20, 80]} min={0} max={100} />);
+
+    const sliderElement = document.querySelector(".rc-slider");
+    expect(sliderElement).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx]: Thanks. This has been added.

File: web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+});
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/RangeSlider/index.test.tsx]: This test doesn't handle onChange well so will keep it as is.

File: web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    );
+
+    expect(screen.getByTestId("custom-trigger")).toBeInTheDocument();
+  });
+
+  test("renders tips text when no icon or trigger component is provided", () => {
+    render(<Tooltip type="custom" text="Help text" />);
+
+    expect(screen.getByText("TIPS")).toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx]: Thanks. This will be added.

File: web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+import { describe, test, expect } from "vitest";
+
+import Tooltip from "./index";
+
+describe("Tooltip Component", () => {
+  test("renders experimental tooltip with default flask icon", () => {
+    render(<Tooltip type="experimental" />);
+
+    const icon = document.querySelector("svg");
+    expect(icon).toBeInTheDocument();
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx]: This strategy is okay given the current implementation for svgs

File: web/src/beta/lib/reearth-ui/components/Selector/index.test.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
+import { screen, fireEvent, render } from "@reearth/test/utils";
+import { describe, test, expect, vi } from "vitest";
+
+import { Selector } from "./index";
+
+global.ResizeObserver = class ResizeObserver {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+};
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Selector/index.test.tsx]: Thanks. I've removed this global reference as its in `test/utils.tsx`

File: web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+
+    expect(screen.getByTestId("custom-trigger")).toBeInTheDocument();
+  });
+
+  test("renders tips text when no icon or trigger component is provided", () => {
+    render(<Tooltip type="custom" text="Help text" />);
+
+    expect(screen.getByText("TIPS")).toBeInTheDocument();
+  });
+});
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx]: Thanks. This has been added.

File: web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    render(<Tooltip type="custom" text="Help information" />);
+
+    const trigger = screen.getByText("TIPS");
+    await fireEvent.click(trigger);
+
+    expect(screen.getByText("Help information")).toBeInTheDocument();
+
+    await fireEvent.click(trigger);
+    expect(screen.queryByText("Help information")).not.toBeInTheDocument();
+  });
```
[the remark this answers is not in the document]
Comment (mulengawilfred) [reply web/src/beta/lib/reearth-ui/components/Tooltip/index.test.tsx]: Thanks. Renamed the test to handle click events.

**Review by mkumbobeaty — APPROVED** [review]
